docs/userguide/perturbative_solvers.rst

```diff
@@ -0,0 +1,308 @@
+    single solver run than typical solvers, and thus it is circumstance-specific whether
+    the trade-off between speed of a single run and resource consumption is advantageous.
+    Due to the parallelized nature, the comparison of execution times demonstrated in this
+    userguide are highly hardware-dependent.
+
+
+In this tutorial we use a simple transmon model:
+
+.. math:: H(t) = 2 \pi \nu N + \pi \alpha N(N-I) + s(t) \times 2 \pi r (a + a^\dagger)
+
+where:
+
+-  :math:`N`, :math:`a`, and :math:`a^\dagger` are, respectively, the
+   number, annihilation, and creation operators.
+-  :math:`\nu` is the qubit frequency and :math:`r` is the drive
+   strength.
+-  :math:`s(t)` is the drive signal, which we will take to be on
+   resonance with envelope :math:`f(t) = A \frac{4t (T - t)}{T^2}`
+   for a given amplitude :math:`A` and total time :math:`T`.
+
+We will walk through the following steps:
+
+1. Configure ``qiskit-dynamics`` to work with JAX.
+2. Construct the model.
+3. How-to construct and simulate using the Dyson-based perturbative solver.
+4. Simulate using a traditional ODE solver, comparing speed.
+5. How-to construct and simulate using the Magnus-based perturbative solver.
+
+1. Configure to use JAX
+-----------------------
+
+These simulations will be done with JAX array backend to enable
+compilation. See the user guide entry on using JAX for a detailed
+breakdown of using JAX with dynamics.
+
+.. jupyter-execute::
+
+    from qiskit_dynamics.array import Array
+
+    # configure jax to use 64 bit mode
+    import jax
+    jax.config.update("jax_enable_x64", True)
+
+    # tell JAX we are using CPU
+    jax.config.update('jax_platform_name', 'cpu')
+
+    # set default backend
+    Array.set_default_backend('jax')
+
+2. Construct the model
+----------------------
+
+First, construct the model described in the introduction. Here we use a
+higher dimension to observe a difference between the solvers.
+
+.. jupyter-execute::
+
+    import numpy as np
+
+    dim = 10
+
+    v = 5.
+    anharm = -0.33
+    r = 0.02
+
+    a = np.diag(np.sqrt(np.arange(1, dim)), 1)
+    adag = np.diag(np.sqrt(np.arange(1, dim)), -1)
+    N = np.diag(np.arange(dim))
+
+    # static part
+    static_hamiltonian = 2 * np.pi * v * N + np.pi * anharm * N * (N - np.eye(dim))
+    # drive term
+    drive_hamiltonian = 2 * np.pi * r * (a + adag)
+
+    # total simulation time
+    T = 1. / r
+
+    # envelope function
+    envelope_func = lambda t: t * (T - t) / (T**2 / 4)
+
+3. How-to construct and simulate using the Dyson-based perturbative solver
+--------------------------------------------------------------------------
+
+Constructing the Dyson-based perturbative solver requires specifying several
+configuration parameters, as well as specifying the structure of the
+differential equation more explicitly than using the standard
+:class:`.Solver` object in qiskit-dynamics, which automatically builds
```

Comment:
Suggested rewording/ordering
```
Setting up a :class:.`DysonSolver` requires more setup han the standard
:class:`.Solver` as the user must specify several configuration parameters,
along with the structure of the differential equation:

- A :class:`.DysonSolver` requires direct specification of the LMDE to the
  solver. If we are simulating the Schrodinger equation, we need to
  multiply the Hamiltonian terms by ``-1j`` when passing in as LMDE
  operators.
- The :class:`.DysonSolver` is a fixed step solver, with the step size
  being fixed at instantiation. This step size must be chosen in conjunction
  with the ``expansion_order``, to ensure that a suitable accuracy is attained.
- Over each fixed time-step the :class:`.DysonSolver` solves by computing a
   truncated perturbative expansion.
  - To compute the truncated perturbative expansion, the signal envelopes are
    approximated as a linear combination of Chebyshev polynomials.
  - The order of the Chebyshev approximations, along with central carrier frequencies
    for defining the “envelope” of each ``Signal``, must be provided at instantiation.

See the :ref:`API docs <.DysonSolver>` for a more details.
```

Reply:
Implemented this suggestion with a few edits.